Support rook-pivoting in bkfact

timholy opened the pull request:
LAPACK 3.5 supports alternative [rook-pivoting algorithms](http://www.netlib.org/lapack/improvement.html) for Bunch-Kaufman. This PR wraps this functionality.

Best reviewed with `?w=1` appended to the URL.

andreasnoack commented on base/linalg/lapack.jl:
```diff
@@ -3847,28 +3869,36 @@ for (syconv, hesv, hetrf, hetri, hetrs, elty, relty) in
-     (:csysv_,:csytrf_,:csytri_,:csytrs_,:Complex64, :Float32))
-    @eval begin
+for suffix in (symbol(""), :_rook)
+    for (sysv, sytrf, sytri, sytrs, elty, relty) in
```
I think it would be simpler to have a loop for each function given the convoluted code just below.

andreasnoack commented:
Looks good to me except for the small style comment, maybe we should wait until https://github.com/JuliaLang/julia/pull/14309 has been merged.

timholy committed:
Support rook-pivoting in bkfact

timholy commented:
Rebased and updated, including the style change. Curiously, Travis never ran this as a PR (it did run the push), but AppVeyor did, so hopefully we're OK.

nalimilan commented:
It would be nice to make a bit more advertisement in the future when raising dependency requirements like that. It took me some time to realize why my EPEL RPM builds where failing.

timholy commented:
Since I always build openblas from source, I didn't think about this as a potential problem. Sorry about that, and thanks for fixing!

nalimilan commented:
You're welcome @timholy.

ViralBShah commented:
Could this have been wrapped in a test for `laver()`? That would allow us to avoid making the minimum LAPACK at 3.5. The main reason is that users of MKL may not have new versions of LAPACK.

ViralBShah commented:
Seems like `LAPACK.VERSION[]` is the new way to do this.

ViralBShah commented:
I seem to recollect that even `dgesdd` was introduced in 3.5 or perhaps 3.4, and used to be protected with a version check once upon a time.

BTW, I am ok with 3.5 being the minimum, but just wanted to point out the issues.

eschnett commented:
One can probably work around missing LAPACK functions in MKL with some linker magic. One needs to link against MKL explicitly (i.e. using an `-l` flag, not the `-mkl` flag), then link against a new LAPACK. This will preferably resolve symbols against MKL, and then take any missing symbols from LAPACK.
